cmd/dlt/accountroles/cmd.go

```diff
@@ -364,26 +367,29 @@ func buildCommand(roleNames []string, policyMap map[string][]aws.PolicyDetail,
 			Build()
 		commands = append(commands, deleteRole)
 
-		if len(hcpSharedVpcPoliciesOutput) > 0 { // Delete HCP shared VPC policies
-			for _, hcpSharedVpcPolicy := range hcpSharedVpcPoliciesOutput {
-				hasRhManagedTag := false
-				hasHcpSharedVpcTag := false
-				for _, tag := range hcpSharedVpcPolicy.Policy.Tags {
-					if *tag.Key == tags.RedHatManaged {
-						hasRhManagedTag = true
-					} else if *tag.Key == tags.HcpSharedVpc {
-						hasHcpSharedVpcTag = true
-					}
-				}
-				if hasHcpSharedVpcTag && hasRhManagedTag {
-					deletePolicy := awscb.NewIAMCommandBuilder().
-						SetCommand(awscb.DeletePolicy).
-						AddParam(awscb.PolicyName, *hcpSharedVpcPolicy.Policy.PolicyName).
-						Build()
-					commands = append(commands, deletePolicy)
+		// Delete HCP shared VPC policies
+		for _, hcpSharedVpcPolicy := range hcpSharedVpcPoliciesOutput {
+			hasRhManagedTag := false
+			hasHcpSharedVpcTag := false
+			for _, tag := range hcpSharedVpcPolicy.Policy.Tags {
+				if *tag.Key == tags.RedHatManaged {
+					hasRhManagedTag = true
+				} else if *tag.Key == tags.HcpSharedVpc {
+					hasHcpSharedVpcTag = true
 				}
 			}
+			if hasHcpSharedVpcTag && hasRhManagedTag {
+				deletePolicy := awscb.NewIAMCommandBuilder().
+					SetCommand(awscb.DeletePolicy).
+					AddParam(awscb.PolicyName, *hcpSharedVpcPolicy.Policy.PolicyName).
+					Build()
+				hcpSharedVpcPolicyCommands[*hcpSharedVpcPolicy.Policy.PolicyName] = deletePolicy
+			}
 		}
 	}
+
+	for _, command := range hcpSharedVpcPolicyCommands {
```

Comment:
Does the order of the commands matter? If it does it shouldn't be a map

Reply:
Order should not matter, deleting the policies can be done in any order I would think since they can only be deleted if they are not attached to anything, and the commands are printed last. the policies have no connection to each other